packages/components/src/InputText/InputText.tsx

```diff
@@ -50,14 +61,54 @@ function InputTextInternal(
     },
   }));
 
+  const [currentRows, setRows] = useState(initializeRows());
+
   return (
     <FormField
+      {...props}
       type={props.multiline ? "textarea" : "text"}
       ref={inputRef}
-      {...props}
+      onChange={handleChange}
+      rows={currentRows}
     />
   );
 
+  function initializeRows() {
+    if (!props.rows) return;
+    if (typeof props.rows === "number") {
+      return props.rows;
+    } else {
+      return props.rows.min;
+    }
+  }
+
+  function handleChange(newValue: string) {
+    props.onChange && props.onChange(newValue);
+    if (!props.rows) return;
+    if (
+      typeof props.rows !== "number" &&
+      inputRef &&
+      inputRef.current instanceof HTMLTextAreaElement
+    ) {
+      const currentRef = inputRef.current as HTMLTextAreaElement;
+      const textareaLineHeight = parseInt(
+        window.getComputedStyle(currentRef).getPropertyValue("line-height"),
+        10,
+      );
+      const textareaScrollHeight = currentRef.scrollHeight;
+      const calculatedRows = calculateRows(
+        textareaScrollHeight,
+        textareaLineHeight,
+      );
+      if (
+        calculatedRows >= props.rows.min &&
+        calculatedRows <= props.rows.max
+      ) {
+        setRows(calculatedRows);
+      }
```

Comment:
This whole block could be a separate function. If you define it on line 111 (i.e. within the component scope) then it will have access to all of the scope that it currently does, so it shouldn't be too complicated.

Reply:
Basically, you could pass in `inputRef` and return `calculatedRows` which you would then use in `setRows(calculatedRows)`.
Happy to pair on that change if you want!

Reply:
Chatted about this separately and got it resolved :)